lfpse.R

```diff
@@ -6,62 +6,64 @@ if(lfpse_categorical==0){
   lfpse_categorical <- expr(1==1)
 }
 
-# reading reference tables
-QuestionReference <- tbl(con_lfpse, "QuestionReference") |> collect()
-ResponseReference <- tbl(con_lfpse, "ResponseReference") |> collect()
-
+#set latest revision table from events
+latest_revision_table <- tbl(con_lfpse, in_schema("analysis", "Events")) |>
+  group_by(Reference) |>
+  summarise(Revision = max(Revision), #i.e., the most up to date version of a record 
+            reported_date = min(SubmissionDate)
+  ) |>
+  ungroup() 
 
+#gather analysis tables
 analysis_table_names <- c(
   "Metadata_Responses",
-  "Incident_Responses",
+  "Incident_Responses", 
   "Risk_Responses",
   "Outcome_Responses",
   "GoodCare_Responses",
   "EventDetails_Responses",
   "EventTime_Responses",
   "Location_Responses",
-  "Patient_Responses",
+  "Events",
+  "Patient_Responses", 
   "Medication_Responses",
   "Devices_Responses",
   "Reporter_Responses",
   "Governance_Responses" # ,
   # "Findings_Responses"#, 
   # "DmdMedication_Responses"
-)
-
-#create table for all tables except events
-analysis_tables_non_event <- lapply(analysis_table_names, function(x){
-  tbl(con_lfpse, in_schema("analysis", x)) 
-})
+  )
 
-#create table for events, filter for most recent revision
-event_table<- tbl(con_lfpse, in_schema("analysis","Events")) %>%
-  group_by(Reference) |>
-  mutate(reported_date = min(SubmissionDate),
-         max_revision = max(Revision)) %>%
-  ungroup() %>%
-  filter(Revision == max_revision)
+#bring all tables together
+analysis_tables <- lapply(analysis_table_names, function(x){
+  tbl(con_lfpse, in_schema("analysis", x)) }
+)
 
-#combine event table and non event tables into one
-analysis_tables<- c(list(event_table), analysis_tables_non_event)
+lfpse_analysis_tables <- c(list(latest_revision_table), analysis_tables)
 
 # duplicates will be present due to inclusion of Patient_Responses which is one row per patient (EntityId)
-lfpse_parsed <- reduce(analysis_tables, left_join, by = c("Reference", "Revision")) |>
-  rename(occurred_date = T005,
-         revision_date = SubmissionDate) |>
+lfpse_parsed <- reduce(lfpse_analysis_tables, 
+                       left_join, by = c("Reference", "Revision")) |>
+  rename(occurred_date = T005) |>
   # a conversion factor from days will be needed here, but appears to be DQ issues
   # suggest we wait for resolution before converting from days to years
-  mutate(P004_years = as.numeric(P004))
+  mutate(P004_days = as.numeric(P004))
 
 #sql_render(lfpse_parsed) this is a useful step to check the SQL has rendered sensibly
 
-lfpse_filtered_categorical<- lfpse_parsed|>
+#record time to keep track of query speeds
+tic <- Sys.time()

```

Comment:
this is already in flow.R - suggest separating by database if you wanted to time the different elements

Reply:
It is, but I thought that was a clock start for the entire search process. That would mean that for requests with an NRLS search, the clock would start running when nrls.R is sourced. What is proposed here is to time the data collection step particular to LFPSE. This is more for us to keep an eye out in case speed issues re emerge.

Reply:
it is, but it's also called `tic`, so would be overwritten here - so either get rid of the one in flow, or separate into tic_nrls etc. It shouldn't make much difference since, if search_nrls = T, nrls.R is skipped, and that eval shouldn't take long. You can test it :)

Reply:
Of course they would be overwritten! I've created `tic_nrls` and `tic_lfpse` objects :)